app/services/idv/phone_step.rb

```diff
@@ -43,7 +43,13 @@ def async_state_done(async_state)
       @idv_result = async_state.result
 
       success = idv_result[:success]
-      handle_successful_proofing_attempt if success
+      if success
+        handle_successful_proofing_attempt
+      else
+        unless failure_reason == :timeout
+          idv_session.add_failed_phone_step_number(idv_session.previous_phone_step_params[:phone])
+        end
+      end
```

Comment:
Can we flatten with an `elsif` ?

```suggestion
      elsif failure_reason != :timeout
        idv_session.add_failed_phone_step_number(idv_session.previous_phone_step_params[:phone])
      end
```

Or maybe extract a "handler" method to complement the success handler?

```suggestion
      else
        handle_failed_proofing_attempt
      end

      def handle_failed_proofing_attempt
        return if failure_reason != :timeout
        idv_session.add_failed_phone_step_number(idv_session.previous_phone_step_params[:phone])
      end
```

Reply:
i like the handler suggestion but will modify the operator ... thanks